### PR TITLE
#860 switch to using assertj instead of FEST assertions

### PR DESCRIPTION
**Pull request by filiphr**
This is a pull request for (#860) switching the assertions framework to the maintained assertj instead of FEST.

### CODE REVIEW

File: processor/src/test/java/org/mapstruct/ap/testutil/assertions/JavaFileAssert.java [hunk 1]
```diff
@@ -44,15 +44,16 @@ public JavaFileAssert(File actual) {
...
+    public AbstractCharSequenceAssert<?, String> content() {
         exists();
         isFile();
 
         try {
             return Assertions.assertThat( Files.toString( actual, Charsets.UTF_8 ) );
         }
         catch ( IOException e ) {
-            failIfCustomMessageIsSet( e );
+            //TODO shouldn't we just fail if something like this happens.
```
Comment (filiphr) [inline processor/src/test/java/org/mapstruct/ap/testutil/assertions/JavaFileAssert.java]: @gunnarmorling Have a better look at this TODO. `failIfCustomMessageIsSet` does not exists in assertj anymore and I was not sure whether to directly fail with some message or do something else. Previously no custom message was set so even if an exception occurred the test did not fail.

Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/testutil/assertions/JavaFileAssert.java]: Good catch! Let's just use `fail("Unable to read " + actual, e)`.

**Comment by agudian** [comment]
Awesome, @filiphr, thanks for the PR! Would you replace the TODO with the `fail` call? You can directly amand your commit or squash the change to force-push it to your branch.

**Comment by gunnarmorling** [comment]
Nice one, thanks a lot!

**Comment by filiphr** [comment]
I am glad that you like the PR. I'll fix the TODO and do a force-push, tomorrow evening probably

**Comment by filiphr** [comment]
@agudian I replaced the TODO with a fail message

**Comment by gunnarmorling** [comment]
@filiphr I'm getting some failures in the integration tests, supposedly due to the requirement of JDK 8 by that new version (we run some integration tests on previous compiler versions). Didn't look into the details yet, but it seems we may actually have to use that version that run with JDK 7. Did you get the same when running `mvn clean install` on the entire project?

**Comment by gunnarmorling** [comment]
Jenkins, test this please.

**Comment by filiphr** [comment]
@gunnarmorling I did mvn clean install on the entire project. I did it again and realized that it didn't run with oracle_java_6 and oracle_java_7 that is why I didn't see the errors. A solution would be to use the 1.x branch which is requires java 6 or above. 2.x is for java 7 or higher.

Is it an option to use 1.x only in the integration tests?

**Comment by filiphr** [comment]
My last commit should fix the failing tests

**Comment by agudian** [comment]
ok to test

**Comment by agudian** [comment]
I think it's a good idea to use the Java 6 compatible version only in the integration tests. Focus there isn't on fancy assertions anyway, but on correct compilation using the different compilers.

**Comment by agudian** [comment]
Alright, the pull-request-builder looks healthy now as well. I'd say you squash the second commit again into the first one and I'll push it to master then... :+1:

**Comment by agudian** [comment]
Never mind, forgot we have squash-and-merge for quite some time now here... :neckbeard:

**Pull request merged**